Makefile

```diff
@@ -5,13 +5,14 @@ export SHELLOPTS:=$(if $(SHELLOPTS),$(SHELLOPTS):)pipefail:errexit
 MAKEFLAGS += --no-builtin-rules
 .SUFFIXES:
 
-BUILD_DATE             = $(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
-GIT_COMMIT             = $(shell git rev-parse HEAD)
-GIT_REMOTE             = origin
-GIT_BRANCH             = $(shell git rev-parse --symbolic-full-name --verify --quiet --abbrev-ref HEAD)
-GIT_TAG                = $(shell git describe --exact-match --tags --abbrev=0  2> /dev/null || echo untagged)
-GIT_TREE_STATE         = $(shell if [ -z "`git status --porcelain`" ]; then echo "clean" ; else echo "dirty"; fi)
-DEV_BRANCH             = $(shell [ $(GIT_BRANCH) = master ] || [ `echo $(GIT_BRANCH) | cut -c -8` = release- ] || [[ "$(GIT_TAG)" =~ ^v[0-9]+\.[0-9]+\.[0-9]+.*$$ ]] && echo false || echo true)
+BUILD_DATE            := $(shell date -u +'%Y-%m-%dT%H:%M:%SZ')
+GIT_COMMIT            := $(shell git rev-parse HEAD)
+GIT_REMOTE            := origin
+GIT_BRANCH            := $(shell git rev-parse --symbolic-full-name --verify --quiet --abbrev-ref HEAD)
+GIT_TAG               := $(shell git describe --exact-match --tags --abbrev=0  2> /dev/null || echo untagged)
+GIT_TREE_STATE        := $(shell if [ -z "`git status --porcelain`" ]; then echo "clean" ; else echo "dirty"; fi)
+RELEASE_TAG           := $(shell if [[ "$(GIT_TAG)" =~ ^v[0-9]+\.[0-9]+\.[0-9]+.*$$ ]]; then echo "true"; else echo "false"; fi)
+DEV_BRANCH            := $(shell [ $(GIT_BRANCH) = master ] || [ `echo $(GIT_BRANCH) | cut -c -8` = release- ] || [ $(RELEASE_TAG) = true ] && echo false || echo true)
```

Comment:
The `:=` is _very_ important. With just `=`, the `$(shell ...` script is run every time the variable is read. See this test:

```
# This perl script returns current time in milliseconds 
BUILD_DATE            = $(shell perl -MTime::HiRes -e 'printf("%.0f\n",Time::HiRes::time()*1000)')
$(info BUILD_DATE=$(BUILD_DATE))
$(info BUILD_DATE=$(BUILD_DATE))
```
```
$ make
BUILD_DATE=1615332076108
BUILD_DATE=1615332076127
```
See the same variable with two different values

Reply:
On the other hand:
```
# This perl script returns current time in milliseconds 
BUILD_DATE            := $(shell perl -MTime::HiRes -e 'printf("%.0f\n",Time::HiRes::time()*1000)')
$(info BUILD_DATE=$(BUILD_DATE))
$(info BUILD_DATE=$(BUILD_DATE))
```
```
$ make
BUILD_DATE=1615332217826
BUILD_DATE=1615332217826
```